### PR TITLE
[receiver/vcenter]Fix x509 with TLS.InsecureSkipVerify setting

### PR DESCRIPTION
**Pull request by wzy531a**
**Description:** <Describe what has changed.>
Fix x509 with TLS.InsecureSkipVerify setting

**Link to tracking Issue:** <Issue number if applicable>
[issues15701](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/15701)

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by djaglowski — COMMENTED** [review]
@schmikei, please take a look.

File: receiver/vcenterreceiver/scraper_test.go [hunk 1]
```diff
@@ -32,14 +33,37 @@ import (
...
+		TLSClientSetting: configtls.TLSClientSetting{
+			Insecure:           true,
+			InsecureSkipVerify: true,
+		},
+	}
+
+	testScrape(t, ctx, cfg)
+}
+
+func testScrape(t *testing.T, ctx context.Context, cfg *Config) {
```
Comment (schmikei) [inline receiver/vcenterreceiver/scraper_test.go:66]:
```suggestion
func testScrape(ctx context.Context, t *testing.T, cfg *Config) {
```

Linter yells at you for
> scraper_test.go:66:31: context-as-argument: context.Context should be the first parameter of a function (revive)
func testScrape(t *testing.T, ctx context.Context, cfg *Config) {

File: receiver/vcenterreceiver/scraper_test.go [hunk 2]
```diff
@@ -32,14 +33,37 @@ import (
...
+		Endpoint: mockServer.URL,
+		Username: mock.MockUsername,
+		Password: mock.MockPassword,
+		TLSClientSetting: configtls.TLSClientSetting{
+			Insecure:           true,
+			InsecureSkipVerify: true,
+		},
+	}
+
+	testScrape(t, ctx, cfg)
```
Comment (schmikei) [inline receiver/vcenterreceiver/scraper_test.go:63]:
```suggestion
	testScrape(ctx, t, cfg)
```

Fixing ordering due to convention, see later comment.

File: receiver/vcenterreceiver/scraper_test.go [hunk 3]
```diff
@@ -32,14 +33,37 @@ import (
...
+	mockServer := mock.MockServer(t, false)
 
 	cfg := &Config{
 		Metrics:  metadata.DefaultMetricsSettings(),
 		Endpoint: mockServer.URL,
 		Username: mock.MockUsername,
 		Password: mock.MockPassword,
 	}
+
+	testScrape(t, ctx, cfg)
```
Comment (schmikei) [inline receiver/vcenterreceiver/scraper_test.go:45]:
```suggestion
	testScrape(ctx, t, cfg)
```

Fixing ordering due to lint, see later comment.

File: receiver/vcenterreceiver/internal/mockserver/client_mock.go [hunk 4]
```diff
@@ -45,8 +45,8 @@ type soapEnvelope struct {
 }
 
 // MockServer has access to recorded SOAP responses and will serve them over http based off the scraper's API calls
-func MockServer(t *testing.T) *httptest.Server {
-	vsphereMock := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+func MockServer(t *testing.T, useTls bool) *httptest.Server {
```
Comment (schmikei) [inline receiver/vcenterreceiver/internal/mockserver/client_mock.go:48]:
```suggestion
func MockServer(t *testing.T, useTLS bool) *httptest.Server {
```

linter wants this
> internal/mockserver/client_mock.go:48:31: var-naming: func parameter useTls should be useTLS (revive)
func MockServer(t *testing.T, useTls bool) *httptest.Server {

File: receiver/vcenterreceiver/internal/mockserver/client_mock.go [hunk 5]
```diff
@@ -69,8 +69,13 @@ func MockServer(t *testing.T) *httptest.Server {
 		w.WriteHeader(http.StatusOK)
 		w.Header().Set("Content-Type", "text/xml")
 		_, _ = w.Write(body)
-	}))
-	return vsphereMock
+	})
+
+	if useTls {
```
Comment (schmikei) [inline receiver/vcenterreceiver/internal/mockserver/client_mock.go:74]:
```suggestion
	if useTLS {
```

**Review by schmikei — COMMENTED** [review]
Thanks for the fix! 

I think there may be some extending issues with the integration test that I can help you fix, I think a later version of govmomi introduced some disk information that was not recorded before.

@djaglowski could you perhaps enable workflows for this PR to make sure?

File: receiver/vcenterreceiver/scraper_test.go [hunk 6]
```diff
@@ -16,6 +16,7 @@ package vcenterreceiver // import github.com/open-telemetry/opentelemetry-collec
 
 import (
 	"context"
+	"go.opentelemetry.io/collector/config/configtls"
```
Comment (schmikei) [inline receiver/vcenterreceiver/scraper_test.go:19]: This needs to go into the second import order, other than that, I think it looks good.

Comment (wzy531a) [reply receiver/vcenterreceiver/scraper_test.go]: I have removed "go.opentelemetry.io/collector/config/configtls" in commit [e138022](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/15704/commits/e13802243a68167908ed6cba72bcc348f1a39fc3).

**Review by schmikei — APPROVED** [review]
@wzy531a thanks for the fix!

@djaglowski I think this looks good to me

**Review by djaglowski — APPROVED** [review]

**Comment by djaglowski** [comment]
Thanks @wzy531a

**Pull request merged**